src/API/VendorAPI.php

```diff
@@ -78,22 +92,48 @@ public function get_version() {
 	 * @since 1.0.0
 	 */
 	public function register_routes() {
+		if ( $this->supports_multiple_endpoints ) {
+			$this->register_router_multiple_methods();
+		} else {
+			$this->register_router_single_method();
+		}
+	}
 
+	/**
+	 * Register endpoint route with single method
+	 *
+	 * @param string $methods The endpoint's methods.
+	 * @param string $endpoint_callback The endpoint's callback.
+	 *
+	 * @since 1.0.11
+	 */
+	public function register_router_single_method( $methods = '', $endpoint_callback = '' ) {
 		$namespace = $this->api_namespace . $this->api_version;
 
 		register_rest_route(
 			$namespace,
 			'/' . $this->base,
 			array(
 				array(
-					'methods'             => $this->methods,
-					'callback'            => array( $this, $this->endpoint_callback ),
+					'methods'             => empty( $methods ) ? $this->methods : $methods,
+					'callback'            => array( $this, empty( $endpoint_callback ) ? $this->endpoint_callback : $endpoint_callback ),
```

Comment:
Minor: Determining the value of `$endpoing_callback` should happen in the initial section of the method, separately from this function call.